RIOT-OS/RIOT
# tests/xtimer_mutex_lock_timeout: cleanup python automated test

### Contribution description

This PR is a small clean of the `tests/xtimer_mutex_lock_timeout` Python test script: it uses `expect_exact` where appropriate and declare the expect regexp as raw string to avoid the flake8 error `tests/sys/xtimer_mutex_lock_timeout/tests/01-run.py:32:30: W605 invalid escape sequence '\d'`




### Testing procedure

- A green Murdock
- Check the automated test script is still working:

<details>

```
$ make -C tests/xtimer_mutex_lock_timeout/ all test --no-print-directory 
Building application "tests_xtimer_mutex_lock_timeout" for "native" with MCU "native".

"make" -C /work/riot/RIOT/boards/native
"make" -C /work/riot/RIOT/boards/native/drivers
"make" -C /work/riot/RIOT/core
"make" -C /work/riot/RIOT/cpu/native
"make" -C /work/riot/RIOT/cpu/native/periph
"make" -C /work/riot/RIOT/cpu/native/stdio_native
"make" -C /work/riot/RIOT/drivers
"make" -C /work/riot/RIOT/drivers/periph_common
"make" -C /work/riot/RIOT/sys
"make" -C /work/riot/RIOT/sys/auto_init
"make" -C /work/riot/RIOT/sys/div
"make" -C /work/riot/RIOT/sys/shell
"make" -C /work/riot/RIOT/sys/test_utils/interactive_sync
"make" -C /work/riot/RIOT/sys/xtimer
/usr/bin/ld: /work/riot/RIOT/tests/xtimer_mutex_lock_timeout/bin/native/cpu/tramp.o: warning: relocation against `_native_saved_eip' in read-only section `.text'
/usr/bin/ld: warning: creating DT_TEXTREL in a PIE
   text	   data	    bss	    dec	    hex	filename
  44462	    692	  56060	 101214	  18b5e	/work/riot/RIOT/tests/xtimer_mutex_lock_timeout/bin/native/tests_xtimer_mutex_lock_timeout.elf


/work/riot/RIOT/tests/xtimer_mutex_lock_timeout/bin/native/tests_xtimer_mutex_lock_timeout.elf  
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2021.01-devel-568-g58de8-pr/tests/xtimer_mutex_lock_timeout_cleanup)
Starting shell...
> 
> 
> help
mutex_timeout_long_unlocked
help
Command              Description
---------------------------------------
mutex_timeout_long_unlocked unlocked mutex (no-spin timeout)
mutex_timeout_long_locked locked mutex (no-spin timeout)
mutex_timeout_long_locked_low lock low-prio-locked-mutex from high-prio-thread (no-spin timeout)
mutex_timeout_short_unlocked unlocked mutex (spin timeout)
mutex_timeout_short_locked locked mutex (spin timeout)
> mutex_timeout_long_unlocked
starting test: xtimer mutex lock timeout
OK

> mutex_timeout_long_locked
mutex_timeout_long_locked
starting test: xtimer mutex lock timeout
OK

> mutex_timeout_long_locked_low
mutex_timeout_long_locked_low
starting test: xtimer mutex lock timeout with thread
threads = 2
THREAD low prio: start
MAIN THREAD: calling xtimer_mutex_lock_timeout
OK
threads = 3
MAIN THREAD: waiting for created thread to end
THREAD low prio: exiting low
threads = 2

> mutex_timeout_short_locked
mutex_timeout_short_locked
starting test: xtimer mutex lock timeout with short timeout and locked mutex
OK

> mutex_timeout_short_unlocked
mutex_timeout_short_unlocked
starting test: xtimer mutex lock timeout with short timeout and unlocked mutex
OK

>
```

</details>




### Issues/PRs references

Raised when running the static test locally in #15358
